### PR TITLE
Added Path variables in GraphFetch persistence model spec.

### PR DESCRIPTION
**Pull request by ankitamritraj**
## How did you test this change?

- [x] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ActionIndicatorFields.ts [hunk 1]
```diff
@@ -32,9 +33,11 @@ export abstract class V1_DeleteIndicator extends V1_ActionIndicatorFields {
 }
 
 export class V1_DeleteIndicatorForGraphFetch extends V1_DeleteIndicator {
+  deleteFieldPath!: object;
```
Comment (akphi) [inline packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ActionIndicatorFields.ts:36]: can you add the comment like we have in other places? something like

```ts
  /**
   * Studio does not process value specification, they are left in raw JSON form
   *
   * @discrepancy model
   */
```

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Deduplication.ts [hunk 2]
```diff
@@ -36,8 +37,12 @@ export class V1_AnyVersion extends V1_Deduplication {
 export abstract class V1_MaxVersion extends V1_Deduplication {}
 
 export class V1_MaxVersionForGraphFetch extends V1_MaxVersion {
+  versionFieldPath!: object;
```
Comment (akphi) [inline packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Deduplication.ts:40]: same comment, please annotate this discrepancy

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Partitioning.ts [hunk 3]
```diff
@@ -36,8 +37,13 @@ export class V1_NoPartitioning extends V1_Partitioning {
 export abstract class V1_FieldBased extends V1_Partitioning {}
 
 export class V1_FieldBasedForGraphFetch extends V1_FieldBased {
+  partitionFieldPaths!: object[];
+
   override get hashCode(): string {
-    return hashArray([PERSISTENCE_HASH_STRUCTURE.FIELD_BASED_FOR_GRAPH_FETCH]);
+    return hashArray([
+      PERSISTENCE_HASH_STRUCTURE.FIELD_BASED_FOR_GRAPH_FETCH,
+      hashObjectWithoutSourceInformation(this.partitionFieldPaths),
```
Comment (akphi) [inline packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Partitioning.ts:45]: potentially, should we do `this.partitionFieldPaths.map(...)` instead?

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ServiceOutput.ts [hunk 4]
```diff
@@ -27,11 +28,15 @@ export abstract class V1_ServiceOutput implements Hashable {
 }
 
 export class V1_GraphFetchServiceOutput extends V1_ServiceOutput {
+  keys!: object[];
+  path!: object;
```
Comment (akphi) [inline packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ServiceOutput.ts:32]: same comment, please annotate this discrepancy

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ActionIndicatorFields.ts [hunk 5]
```diff
@@ -32,9 +33,11 @@ export abstract class DeleteIndicator extends ActionIndicatorFields {
 }
 
 export class DeleteIndicatorForGraphFetch extends DeleteIndicator {
+  deleteFieldPath!: object;
```
Comment (akphi) [inline packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ActionIndicatorFields.ts:36]: same comment, please annotate this discrepancy

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Deduplication.ts [hunk 6]
```diff
@@ -36,8 +37,12 @@ export class AnyVersion extends Deduplication {
 export abstract class MaxVersion extends Deduplication {}
 
 export class MaxVersionForGraphFetch extends MaxVersion {
+  versionFieldPath!: object;
```
Comment (akphi) [inline packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Deduplication.ts:40]: same comment, please annotate this discrepancy

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Partitioning.ts [hunk 7]
```diff
@@ -36,8 +37,12 @@ export class NoPartitioning extends Partitioning {
 export abstract class FieldBased extends Partitioning {}
 
 export class FieldBasedForGraphFetch extends FieldBased {
+  partitionFieldPaths!: object[];
```
Comment (akphi) [inline packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Partitioning.ts:40]: same comment, please annotate this discrepancy

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ServiceOutput.ts [hunk 8]
```diff
@@ -27,11 +28,15 @@ export abstract class ServiceOutput implements Hashable {
 }
 
 export class GraphFetchServiceOutput extends ServiceOutput {
+  keys!: object[];
```
Comment (akphi) [inline packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ServiceOutput.ts:31]: same comment, please annotate this discrepancy

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ServiceOutput.ts [hunk 9]
```diff
@@ -27,11 +28,15 @@ export abstract class ServiceOutput implements Hashable {
...
+  path!: object;
```
Comment (akphi) [inline packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ServiceOutput.ts:32]: same comment, please annotate this discrepancy

File: packages/legend-extension-dsl-persistence/src/graph-manager/__tests__/DSL_Persistence_Roundtrip.test.ts [hunk 10]
```diff
@@ -70,4 +71,8 @@ test(unitTest('DSL Persistence V2 roundtrip'), async () => {
     TEST_DATA__roundtrip_non_temporal_snapshot_date_time_audit as Entity[],
     pluginManager,
   );
+  await TEST__checkBuildingElementsRoundtrip(
```
Comment (akphi) [inline packages/legend-extension-dsl-persistence/src/graph-manager/__tests__/DSL_Persistence_Roundtrip.test.ts:74]: this test is fine, but honestly, the better and more rigourous test to add is a full grammar roundtrip test in `packages/legend-manual-tests`, could you add one?

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ActionIndicatorFields.ts (hunk 1 above)
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ActionIndicatorFields.ts:36]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Deduplication.ts (hunk 2 above)
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Deduplication.ts:40]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Partitioning.ts [hunk 11]
```diff
@@ -36,8 +37,13 @@ export class V1_NoPartitioning extends V1_Partitioning {
 export abstract class V1_FieldBased extends V1_Partitioning {}
 
 export class V1_FieldBasedForGraphFetch extends V1_FieldBased {
+  partitionFieldPaths!: object[];
```
[the remark this answers is not in the document]
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Partitioning.ts:45]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ServiceOutput.ts [hunk 12]
```diff
@@ -27,11 +28,15 @@ export abstract class V1_ServiceOutput implements Hashable {
 }
 
 export class V1_GraphFetchServiceOutput extends V1_ServiceOutput {
+  keys!: object[];
```
[the remark this answers is not in the document]
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ServiceOutput.ts:36]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ServiceOutput.ts (hunk 4 above)
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ServiceOutput.ts:42]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ActionIndicatorFields.ts (hunk 5 above)
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ActionIndicatorFields.ts:41]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Deduplication.ts (hunk 6 above)
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Deduplication.ts:45]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Partitioning.ts (hunk 7 above)
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Partitioning.ts:45]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ServiceOutput.ts (hunk 8 above)
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ServiceOutput.ts:36]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ServiceOutput.ts (hunk 9 above)
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_ServiceOutput.ts:42]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph-manager/__tests__/DSL_Persistence_Roundtrip.test.ts (hunk 10 above)
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph-manager/__tests__/DSL_Persistence_Roundtrip.test.ts:74]: Sure, have added one test case.

**Review by akphi — APPROVED** [review]

**Pull request merged**